src/Functions/FunctionBinaryArithmetic.h

```diff
@@ -1326,6 +1423,21 @@ class FunctionBinaryArithmetic : public IFunction
             return getReturnTypeImplStatic(new_arguments, context);
         }
 
+        if (isArray(arguments[0]) && isArray(arguments[1]))
+        {
+            DataTypes new_arguments {
+                    static_cast<const DataTypeArray &>(*arguments[0]).getNestedType(),
+                    static_cast<const DataTypeArray &>(*arguments[1]).getNestedType(),
+            };
+
+            return std::make_shared<DataTypeArray>(getReturnTypeImplStatic(new_arguments, context));
+        }
+
+        if (isArray(arguments[0]) || isArray(arguments[1]))
+            throw Exception(ErrorCodes::BAD_ARGUMENTS,
+            "Cannot execute arguments of different type. Type of the first argument: {}, type of the second argument: {}",
+            arguments[0]->getName(), arguments[1]->getName());
+
```

Comment:
Actually this part catches some unwanted scenarios

Reply:
Then my question what happened in those scenarios before your changes? Because as far as I understand your changes only enabled the case the case when the two arguments are arrays and nothing else, thus the exception at the end of this function should be thrown without this check preventing any kind of operation where only one of the arguments is an array.

[Demo](https://play.clickhouse.com/play?user=play#U0VMRUNUIFsxLDJdICsgMzs=).